src/client/game.cpp

```diff
@@ -335,6 +354,29 @@ class GameGlobalShaderConstantSetter : public IShaderConstantSetter
 
 		float saturation = lighting.saturation;
 		m_saturation_pixel.set(&saturation, services);
+		video::SColorf artificial_light = lighting.artificial_light_color;
+		m_artificial_light.set(artificial_light, services);
+
+		float gamma = m_gamma;
+		m_gamma_pixel.set(&gamma, services);
+
+		const Vignette &vignette_params = lighting.vignette;
+		m_vignette_dark_pixel.set(&vignette_params.dark, services);
+		m_vignette_bright_pixel.set(&vignette_params.bright, services);
+		m_vignette_power_pixel.set(&vignette_params.power, services);
+
+		m_foliage_translucency_pixel.set(&lighting.foliage_translucency, services);
+		m_specular_intensity_pixel.set(&lighting.specular_intensity, services);
+
+		if (g_settings->getBool("enable_color_grading")) {
+			const ColorDecisionList& cdl_params = lighting.cdl;
+			core::vector3df slope = cdl_params.slope;
+			m_cdl_slope_pixel.set(slope, services);
+			core::vector3df offset = cdl_params.offset;
+			m_cdl_offset_pixel.set(offset, services);
+			core::vector3df power = cdl_params.power;
+			m_cdl_power_pixel.set(power, services);
```

Comment:
I'm not sure why the existing code does it so much but there's no need to
```suggestion
			m_cdl_slope_pixel.set(cdl_params.slope, services);
			m_cdl_offset_pixel.set(cdl_params.offset, services);
			m_cdl_power_pixel.set(cdl_params.power, services);
```